src/js/components/SidebarComponent.jsx

```diff
@@ -0,0 +1,106 @@
+var Link = require("react-router").Link;
+var React = require("react/addons");
+
+var AppListLabelsFilterComponent =
+  require("../components/AppListLabelsFilterComponent");
+var AppListStatusFilterComponent =
+  require("../components/AppListStatusFilterComponent");
+
+var SidebarComponent = React.createClass({
+  displayName: "SidebarComponent",
+
+  contextTypes: {
+    router: React.PropTypes.func
+  },
+
+  propTypes: {
+    groupId: React.PropTypes.string.isRequired,
+    onChange: React.PropTypes.func.isRequired
+  },
+
+  getInitialState: function () {
+    return {
+      filters: {
+        filterLabels: [],
+        filterStatus: []
+      }
+    };
+  },
+
+  updateFilterLabels: function (filterLabels) {
+    var state = this.state;
+
+    this.setState(Object.assign(state.filters, {
```

Comment:
Seriously question, do I modify state here?
I don't change the object reference from the ```filters```-object, that is bound to the state-object, so I actually don't modify the state. I only modify the filters-object contents.
I am unsure, if this is a state-change, though??